receiver.go

```diff
@@ -273,17 +275,18 @@ func (r *Receiver) sendDisposition(first uint32, last *uint32, state encoding.De
 	return r.link.Session.txFrame(fr, nil)
 }
 
-func (r *Receiver) messageDisposition(ctx context.Context, id uint32, state encoding.DeliveryState) error {
+func (r *Receiver) messageDisposition(ctx context.Context, msg *Message, state encoding.DeliveryState) error {
 	var wait chan error
 	if r.link.ReceiverSettleMode != nil && *r.link.ReceiverSettleMode == ModeSecond {
-		debug(3, "RX: add %d to inflight", id)
-		wait = r.inFlight.add(id)
+		debug(3, "RX: add %d to inflight", msg.deliveryID)
+		wait = r.inFlight.add(msg.deliveryID)
+		defer func() { r.link.DeleteUnsettled(msg) }()
```

Comment:
Here's where we remove the message from the unsettled map.

Reply:
Does the removal of the unsettled message depend on whether 'sendDisposition' below succeeds or fails? Was wondering why it was deferred (ie, which operation are you trying to get it to happen after?)

Reply:
I'm going over the spec and I think this might be wrong.  In section 2.6.12, I believe that we shouldn't be removing the entry from the unsettled map until we have final receipt of settlement from the sender.  i.e. we (the receiver) receive the message, send a disposition frame indicating we accept the message, the service sends a disposition frame acknowledging our acceptance, then we remove it from the unsettled map.  Do you concur?

Reply:
I we're both reading that section the same way. 

1. We send a disposition
2. Sender acks it and sends _us_ a disposition 
3. Once we receive the ACK of our ACK (oof) then we can delete it from our map